### DISCUSSION TITLE
Priority behavior

### DISCUSSION THREAD

**Discussion by DevNoud** [quartznet/quartznet#1838] [Q&A] [answered]
In our usecase we have al lot of jobs scheduled at midnight to do some work.
There is also another job that must be fired every 5 minutes.

The jobs scheduled at midnight prevent the every 5 minutes job now to get fired. Resulting that this job waits until all the midnight jobs are finished before getting fired.
I have tried to change the priority of the 5 minutes job trigger, but as also noted in the docs, this does not fix my problem as the priority is only used to prioritize jobs with the same fire time.

So my questions now are: 
- Can it be done in quartz?
- If yes, how?

**Comment by lahma** [comment, marked as answer]
You could use two separate schedulers. One for mass processing at best effort and another for critical jobs that need to be run under guaranteed processing availability.

**Reply by DevNoud** [reply]
Thnx, that solved my problem.

**Reply by DevNoud** [reply]
Is it possible to start two schedulars in aspnetcore?

If I do the folowing in the startup.cs in the ConfigureServices function, only one schedular seems to be available in my controllers.

```
services.AddQuartz(q =>
{
    q.SchedulerId = "Schedular1";
    q.SchedulerName = "Schedular1";
    q.UseMicrosoftDependencyInjectionJobFactory();
    q.UseSimpleTypeLoader();
    q.UsePersistentStore(store => {
        store.UseClustering();
        store.UseMySql(options => {
            options.ConnectionString = connString;
            options.TablePrefix = "QRTZ_";
        });
        store.UseJsonSerializer();
    });
    q.UseDefaultThreadPool(tp => {
        tp.MaxConcurrency = 10;
    });
}
services.AddQuartz(q =>
{
    q.SchedulerId = "Schedular2";
    q.SchedulerName = "Schedular2";
    q.UseMicrosoftDependencyInjectionJobFactory();
    q.UseSimpleTypeLoader();
    q.UsePersistentStore(store => {
        store.UseClustering();
        store.UseMySql(options => {
            options.ConnectionString = connString;
            options.TablePrefix = "QRTZ_";
        });
        store.UseJsonSerializer();
    });
    q.UseDefaultThreadPool(tp => {
        tp.MaxConcurrency = 2;
    });
}
services.AddQuartzServer(options =>
{
    // when shutting down we want jobs to complete gracefully
    options.WaitForJobsToComplete = true;
});

```

**Reply by lahma** [reply]
Currently the API only allows one scheduler. You could use `SchedulerBuilder` to construct multiple and register them with some helper wrapper to ensure you can access them via DI.
